samcli/local/apigw/local_apigw_service.py

```diff
@@ -234,11 +235,47 @@ def _should_base64_decode_body(binary_types, flask_request, lamba_response_heade
         True if the body from the request should be converted to binary, otherwise false
 
         """
-        best_match_mimetype = flask_request.accept_mimetypes.best_match([lamba_response_headers["Content-Type"]])
+
+        # Get the first part of the content-type header, to allow for extras such as text/html; charset=utf-8
+        content_type = lamba_response_headers['Content-Type'].split(";", 1)[0]
+        best_match_mimetype = flask_request.accept_mimetypes.best_match([content_type])
```

Comment:
How did you validate this is the correct behavior?

Reply:
A MIME type should contain only `type/subtype`.
https://developer.mozilla.org/en-US/docs/Web/HTTP/Basics_of_HTTP/MIME_types

However, the Content-Type header contains the MIME, then potentially extra charset information:
https://www.ietf.org/rfc/rfc2045.txt (page 12)
>  In the Augmented BNF notation of RFC 822, a Content-Type header field
   value is defined as follows:

     content := "Content-Type" ":" type "/" subtype
                *(";" parameter)
                ; Matching of media type and subtype
                ; is ALWAYS case-insensitive.

Passing this extra charset information into `flask_request.accept_mimetypes.best_match` results in Flask returning 500.

Reply:
@martysweet I should have been a little clearer. Does this behavior match API Gateway's? Meaning do they only look at the first Content-Type Header or do they compare with the list of all headers? Did we match documentation or what is observed from the service?

Reply:
@jfuss Ah understood. Looking into it, this might be more complex than just looking at `Content-Type`, though a bit outside of this PR.

So your question is, if multiple Content-Type Headers are returned to API Gateway, which one is used for the comparison against the binary types list?

I can't find anything in the docs, will have to run some tests.

Docs regarding `Accept` and `Content-Type` (which doesn't answer the question).
https://docs.aws.amazon.com/apigateway/latest/developerguide/api-gateway-payload-encodings-configure-with-console.html
https://docs.aws.amazon.com/apigateway/latest/developerguide/api-gateway-payload-encodings-workflow.html